N/nativefiledialog_extended/build_tarballs.jl

```diff
@@ -0,0 +1,102 @@
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/nativefiledialog-extended
+# NATIVE  mingw, apple
+# PORTAL  linux ~armv6l
+# GTK3    linux ~armv6l, freebsd
+# SDL2    linux  armv6l
+CMAKE_FLAGS="-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_CXX_STANDARD=11 \
+    -DCMAKE_C_FLAGS="-Wall -Wextra -pedantic" \
+    -DCMAKE_CXX_FLAGS="-Wall -Wextra -pedantic" \
+    -DBUILD_SHARED_LIBS=ON"
+
+# NATIVE and PORTAL
+BUILD="build"
+PREFIX="${prefix}"
+NFD_PORTAL=OFF
+NFD_APPEND_EXTENSION=OFF
+if [[ "${target}" != *-armv6l-* && "${target}" == *-linux-* ]] ; then
+    NFD_PORTAL=ON
+fi
+if [[ "${target}" != *-armv6l-* && "${target}" == *-linux-* || "${target}" == *-mingw-* || "${target}" == *-apple-* ]] ; then
+    cmake -B ${BUILD} ${CMAKE_FLAGS} \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DNFD_PORTAL=${NFD_PORTAL} \
+        -DNFD_APPEND_EXTENSION=${NFD_APPEND_EXTENSION} \
+        .
+    cmake --build build --parallel ${nproc}
+    cmake --install build
+fi
+
+# GTK3
+BUILD="build_gtk"
+if [[ "${target}" != *-armv6l-* && "${target}" == *-linux-* ]] ; then
+    PREFIX="${prefix}/gtk"
+    mkdir ${PREFIX}
+fi
+NFD_PORTAL=OFF
+NFD_APPEND_EXTENSION=ON
+if [[ "${target}" != *-armv6l-* && "${target}" == *-linux-* || "${target}" == *-freebsd-* ]] ; then
+    cmake -B ${BUILD} ${CMAKE_FLAGS} \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DNFD_PORTAL=${NFD_PORTAL} \
+        -DNFD_APPEND_EXTENSION=${NFD_APPEND_EXTENSION} \
+        .
+    cmake --build ${BUILD} --parallel ${nproc}
+    cmake --install ${BUILD}
+    install ${PREFIX}/lib/libnfd.${dlext} ${libdir}/libnfd_gtk.${dlext}
+    rm -r ${PREFIX}
+else
+    ln -s ${prefix}/lib/libnfd.${dlext} ${libdir}/libnfd_gtk.${dlext}
+fi
+
+# SDL2
+BUILD="build_sdl"
+PREFIX="${prefix}/sdl"
+mkdir ${PREFIX}
+cmake -B ${BUILD} ${CMAKE_FLAGS} \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DNFD_PORTAL=${NFD_PORTAL} \
+    -DNFD_APPEND_EXTENSION=${NFD_APPEND_EXTENSION} \
+    .
+cmake --build ${BUILD} --parallel ${nproc}
+cmake --install ${BUILD}
+install ${PREFIX}/lib/libnfd.${dlext} ${libdir}/libnfd_sdl.${dlext}
+rm -r ${PREFIX}
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libnfd", :libnfd)
+    LibraryProduct("libnfd_gtk", :libnfd_gtk)
+    LibraryProduct("libnfd_sdl", :libnfd_sdl)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency(PackageSpec(name="Xorg_xorgproto_jll", uuid="c4d99508-4286-5418-9131-c86396af500b"); platforms=filter(p->Sys.islinux(p)||Sys.isfreebsd(p), platforms))
+    BuildDependency(PackageSpec(name="GTK3_jll", uuid="77ec8976-b24b-556a-a1bf-49a033a670a6"); platforms=filter(p->Sys.islinux(p)||Sys.isfreebsd(p), platforms))
```

Comment:
Why gtk3 and sdl2 are build-only dependencies?

Reply:
Most of the platforms and users are supported only with "native" (Mingw, Mpple) and "portal" (Linux, everything except `armv6l`) options.

Reply:
I don't quite understand what that means.

Reply:
My plan is to write two package extensions, one for SDL2 and the other for GTK3. In `armv6l` or FreeBSD systems, I plan to show a warning to users; saying that they need to install manually one of the two packages to be able to use this.

Reply:
It looks like I misinterpreted part of the README before. NFDe doesn't have SDL2 backend, only GTK3, Portal (dbus), and Native (Windows, Apple). Dbus now [works](https://github.com/JuliaPackaging/Yggdrasil/pull/8884) for all Linux platforms; then, GTK3 is going to be only an alternative on all supported targets as a package extension.